addons/ondevice-knobs/src/types/Number.js

```diff
@@ -17,33 +17,24 @@ class NumberType extends React.Component {
     super(props);
     this.renderNormal = this.renderNormal.bind(this);
     this.renderRange = this.renderRange.bind(this);
-  }
-
-  numberTransformer = (x) => {
-    if (Number.isNaN(Number(x))) {
-      return x.substr(0, x.length - 1);
-    }
 
-    return x;
-  };
+    const { knob } = this.props;
+    this.defaultValue = knob.value;
```

Comment:
I wasn't expecting it to behave this way but for some reason if you keep changing the TextInput's defaultValue it'll just keep updating like if it was the value prop. That's why I added this, so that defaultValue doesn't change.

Reply:
its probably because the parent component is being re-rendered when it shouldn't

Reply:
I don't think preventing renders is the way to go in this case. I'd like to keep this approach unless you suggest something else.

Reply:
Yeah anyway long term this addon will be deprecated to make way for controls in 6.0. Minimal changes to get things working is fine.